elasticsearch/templates/statefulset.yaml

```diff
@@ -2,25 +2,23 @@
 apiVersion: apps/v1beta1
 kind: StatefulSet
 metadata:
-  name: {{ template "uname" . }}
+  name: {{ include "elasticsearch.fullname" . }}
```

Comment:
This is a very big backwards compatibility change. Changing this will mean that everyone needs to re-deploy their clusters. We would need a very good reason to have to do this outside of a major version change. More discussion around this is happening in https://github.com/elastic/helm-charts/issues/150

Reply:
Let me know how you think we should proceed. I don't mind closing these PRs as they are mostly stylistic in nature.

Reply:
But they do fall in line with what you get now when you run `helm create`... so your call!

Reply:
We are having the same conversations in two PRs :P So I'll just link to my reply from the Kibana PR. https://github.com/elastic/helm-charts/pull/173#discussion_r296128610

> But they do fall in line with what you get now when you run helm create... so your call!

We can consider this for the 8.0 release, but falling in line with the  `helm create` template is not a good enough reason to ask everybody to do a full recreate of all their clusters.